internal/nostr/nostr.go

```diff
@@ -373,32 +408,37 @@ func (svc *Service) handleSubscription(ctx context.Context, subscription *Subscr
 		filter.Until = &until
 	}
 
-	sub, err := relay.Subscribe(ctx, []nostr.Filter{filter})
-	if err != nil {
-		errorChan <- err
-		return
-	}
-	errorChan <- nil
-	go func(){
-		for event := range sub.Events {
-			svc.Logger.WithFields(logrus.Fields{
-				"eventId":   event.ID,
-				"eventKind": event.Kind,
-			}).Infof("received event on subscription %d", subscription.ID)
-			responseEvent := ResponseEvent{
-				SubscriptionId: subscription.ID,
-				NostrId: event.ID,
-				Content: event.Content,
-				RepliedAt: event.CreatedAt.Time(),
+	reconnecting := false
+
+	for {
+		svc.Logger.Infof("subscription %d starting", subscription.ID)
+		sub, err := relay.Subscribe(ctx, []nostr.Filter{filter})
+		if err != nil {
+			if !reconnecting {
+				errorChan <- err
 			}
-			svc.db.Save(&responseEvent)
-			svc.postEventToWebhook(event, subscription.WebhookUrl)
+			return
 		}
-	}()
-	svc.Logger.Infof("subscription %d started", subscription.ID)
-	<-ctx.Done()
-	svc.Logger.Infof("subscription %d closed", subscription.ID)
-	// delete svix app
+		if !reconnecting {
+			errorChan <- nil
+		}
+		err = svc.startSubscription(ctx, subscription, sub)
+		if err != nil {
+			if !reconnecting {
+				reconnecting = true
+			}
+			svc.Logger.WithError(err).Error("error with the relay, reconnecting...")
+			relay, _, err = svc.getRelayConnection(ctx, subscription.RelayUrl)
+			if err != nil {
+				svc.Logger.Infof("error reconnecting to relay: %s", subscription.RelayUrl)
+				svc.Logger.Infof("stopping subscription %d", subscription.ID)
+				svc.stopSubscription(subscription)
```

Comment:
I don't think the subscription should be stopped here - my subscription therefore didn't reconnect

Reply:
Actually I think subscriptions should only be stopped when they are explicitly requested to be stopped through the API? right now we call `stopSubscription` in 4 different places

Reply:
Fixed!